src/internal/scheduled/scheduleIterable.ts

```diff
@@ -22,36 +23,29 @@ export function scheduleIterable<T>(input: Iterable<T>, scheduler: SchedulerLike
         iterator = (input as any)[Symbol_iterator]();
 
         // Schedule the first iteration and emission.
-        subscriber.add(
-          scheduler.schedule(function () {
-            // Check to make sure teardown was not triggered
-            // by the consumer since the last iteration.
-            if (!subscriber.closed) {
-              let value: T;
-              let done: boolean | undefined;
-              try {
-                // Pull the value out of the iterator
-                ({ value, done } = iterator.next());
-              } catch (err) {
-                subscriber.error(err);
-                return;
-              }
-              if (done) {
-                // If it is "done" we just complete. This mimics the
-                // behavior of JavaScript's `for..of` consumption of
-                // iterables, which will not emit the value from an iterator
-                // result of `{ done: true: value: 'here' }`.
-                subscriber.complete();
-              } else {
-                // The iterable is not done, emit the value.
-                subscriber.next(value);
-                // Reschedule. This will cause this function to be
-                // called again on the same scheduled delay.
-                this.schedule();
-              }
+        caughtSchedule(subscriber, scheduler, function () {
+          // Check to make sure teardown was not triggered
+          // by the consumer since the last iteration.
+          if (!subscriber.closed) {
+            let value: T;
+            let done: boolean | undefined;
+            // Pull the value out of the iterator
+            ({ value, done } = iterator.next());
```

Comment:
I think here destructuring could be used (and inferred type is okay?)
```ts
            // Pull the value out of the iterator
            let { value, done } = iterator.next();
```

Reply:
AFAICT, the inference in this suggestion would work if the `iterator` declaration were changed to:

```ts
let iterator: Iterator<T, T>;
```

The second type parameter is necessary - otherwise, it defaults to `any` and the union in `IteratorResult` effects `any` as the inferred type for `value`.

Also, `const` could be used instead of `let`, if the suggestion is accepted.

Reply:
change made.